dns_server.go

```diff
@@ -129,7 +129,12 @@ func dnsMain(l *logrus.Logger, hostMap *HostMap, c *config.C) func() {
 }
 
 func getDnsServerAddr(c *config.C) string {
-	return c.GetString("lighthouse.dns.host", "") + ":" + strconv.Itoa(c.GetInt("lighthouse.dns.port", 53))
+	dnsHost := strings.TrimSpace(c.GetString("lighthouse.dns.host", ""))
+	// Old guidance was to provide the literal `[::]` in `lighthouse.dns.host` but that won't resolve.
+	if dnsHost == "[::]" {
+		dnsHost = "::"
+	}
```

Comment:
Shouldn't this be the other way around? If `dnsHost` is `::`, it should be changed to `[::]`.

Reply:
This is allowing for an incorrect `[::]` to be entered as the address since configs prior to this change would have had to use the incorrect notation for "any ipv6 address", the `net.JoinHostPort()` handles putting the brackets in for us. If we did nothing then the result when the config had `[::]` would be `[[::]]:53` which won't work.

Reply:
Yes that's handled by the line directly underneath this section.

Reply:
Ah okay, that makes sense!